### PR TITLE
V1 Search jobs - Multiple filters

### PR DESCRIPTION
**Pull request by BenChand**
Currently trying to search for "failed jobs", for example, returns a ton of jobs which can be a bit overwhelming when you might only want the ones from the last 2 days. Supporting at most one of each filter seems reasonable.

### CODE REVIEW

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 1]
```diff
@@ -249,7 +249,7 @@ message SearchJobStatusRequest {
         }
     }
 
-    Filter filter = 1;
+    repeated Filter filter = 1;
```
Comment (audunska) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:252]: Will these filters always be anded together? Might be worth clarifying in some way.

Comment (BenChand) [reply cognite/seismic/protos/v1/seismic_service_messages.proto:252]: Yeah they'll always be anded together, I can add a comment

File: cognite/seismic/protos/v1/seismic_service_messages.proto [hunk 2]
```diff
@@ -249,7 +249,9 @@ message SearchJobStatusRequest {
         }
     }
 
-    Filter filter = 1;
+    // Used to remove jobs from the returned stream that do not satisfy *all* of the given constraints.
+    // Note that only one of each filter variant may be applied. For example, attempting to jobs that passed *or* failed will be invalid.
```
Comment (mr-celo) [inline cognite/seismic/protos/v1/seismic_service_messages.proto:253]: "... attempting to jobs that passed ..."

**Review by mr-celo — APPROVED** [review]

File: cognite/seismic/protos/v1/seismic_service_messages.proto (hunk 2 above)
Comment (BenChand) [reply cognite/seismic/protos/v1/seismic_service_messages.proto:253]: Woops

**Comment by mr-celo** [comment]
One question, looking at the Rust side made me wonder, should it a set of optional params instead of a list? Wouldn't that be clearer here and in rust that one of each kind will be taken into account?

**Comment by BenChand** [comment]
Yeah that might make more sense now, I'll have a look at doing that instead

**Pull request merged**